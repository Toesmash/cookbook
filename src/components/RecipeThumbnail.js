import React from 'react';
import { Link } from 'react-router-dom';
import { FaThumbsDown, FaThumbsUp } from 'react-icons/fa'

const RecipeThumbnail = ({ id, title, rating }) => {
    return (
        <Link className='list__item' to={`/view/${id}`}>
            {title.toUpperCase()}
            {rating.like && <span className='list__rating list__rating--up'><FaThumbsUp /></span>}
            {rating.dislike && <span className='list__rating list__rating--down'><FaThumbsDown /></span>}
        </Link>
    );
}

export default RecipeThumbnail;


