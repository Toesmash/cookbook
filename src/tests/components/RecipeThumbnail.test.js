import { shallow } from 'enzyme';
import React from 'react';
import RecipeThumbnail from '../../components/RecipeThumbnail';
import dummyData from '../../redux/fixtures/dummyData';

test('Render RecipeThumbnail', () => {
  const wrapper = shallow(<RecipeThumbnail {...dummyData[0]} />);
  expect(wrapper).toMatchSnapshot();
});