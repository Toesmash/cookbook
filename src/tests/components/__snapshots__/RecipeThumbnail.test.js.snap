// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render RecipeThumbnail 1`] = `
<Link
  className="list__item"
  replace={false}
  to="/view/1cn0agbls"
>
  POTATO DUMPLINGS
  <span
    className="list__rating list__rating--up"
  >
    <Component />
  </span>
</Link>
`;
